rmgpy/rmg/main.py

```diff
@@ -1417,21 +1468,121 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
+                restartSpeciesList = [Species().fromAdjacencyList(adj_list) for adj_list in restartSpeciesList]
+
+                # Load in the restart filter tensors
+                with h5py.File(self.filtersPath, 'r') as f:
+                    try:
+                        unimolecularThreshold_restart = f.get('unimolecularThreshold').value
+                        bimolecularThreshold_restart = f.get('bimolecularThreshold').value
+                        if self.trimolecular:
+                            trimolecularThreshold_restart = f.get('trimolecularThreshold').value
+
+                        # Expand Thresholds to match number of species in the current model.
+                        # Note that we are about to reorder the core species to match the order in the restart seed
+                        # mechanism, so we only need to broadcast to the indices up to numRestartSpcs. Any indices after
+                        # this are additional species that should have `False` for their threshold
+                        unimolecularThreshold = np.zeros(numCoreSpecies, bool)
+                        unimolecularThreshold[:numRestartSpcs] = unimolecularThreshold_restart
+
+                        bimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies), bool)
+                        bimolecularThreshold[:numRestartSpcs, :numRestartSpcs] = bimolecularThreshold_restart
+
+                        if self.trimolecular:
+                            trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
+                            trimolecularThreshold[:numRestartSpcs, :numRestartSpcs, :numRestartSpcs] = \
+                                trimolecularThreshold_restart
+
+                        filters_found = True
+
+                    except KeyError:  # If we can't find the filters then this is because filtering was not used
+                        logging.warning('No filters were found in file {0}. This is to be expected if the restart '
+                                        'files specified are from an RMG job without reaction filtering. Therefore, '
+                                        'RMG will assume that all of the core species from the restart core seed have '
+                                        'already been reacted and will not react them again. Additional species added '
+                                        'to the input file but not in the restart core seed WILL be reacted with the '
+                                        'rest of the core.'.format(self.filtersPath))
+
+                        filters_found = False
+
+                # Reorder the core species to match the indices of the restart filter tensors
+                reordered_core_species = []
+                for spc in restartSpeciesList:
+                    for j, oldCoreSpc in enumerate(self.reactionModel.core.species):
+                        if oldCoreSpc.isIsomorphic(spc):
```

Comment:
I think we should use the `strict=False` argument since we don't generate resonance structures for the restart species.

Reply:
Done